src/imitation/scripts/train_adversarial.py

```diff
@@ -108,25 +109,22 @@ def train(
         return value of `rollout_stats()` on the expert demonstrations loaded from
         `rollout_path`.
     """
+    assert os.path.exists(rollout_path)
     total_timesteps = int(total_timesteps)
 
     tf.logging.info("Logging to %s", log_dir)
     os.makedirs(log_dir, exist_ok=True)
     sacred_util.build_sacred_symlink(log_dir, _run)
 
-    # Calculate stats for expert rollouts. Used for plot and return value.
     expert_trajs = types.load(rollout_path)
-
     if n_expert_demos is not None:
         assert len(expert_trajs) >= n_expert_demos
         expert_trajs = expert_trajs[:n_expert_demos]
 
-    expert_stats = rollout.rollout_stats(expert_trajs)
-
     with networks.make_session():
         if init_tensorboard:
             sb_tensorboard_dir = osp.join(log_dir, "sb_tb")
-            kwargs = init_trainer_kwargs
+            kwargs = copy.deepcopy(init_trainer_kwargs)
```

Comment:
Why is this here? I don't understand the need to have `copy.deepcopy` in script files (here or elsewhere). Non-pickleable objects will cause a problem before it even hits this line of code: they need to be addressed in the code that invokes Ray, not here.

Reply:
The next line modifies `kwargs`, which is illegal without converting to `dict`. I chose to "defensively" deep copy in case for some reason in the future any code downstream (including in the imitation learning algorithm) wants to modify any dicts or lists inside the dict.